generator/lib/code_builders/class_builder.dart

```diff
@@ -16,13 +16,26 @@ Library buildMapperClass(
 
 List<Method> _generateStaticMethods(
     ClassElement abstractClass, Map<String, dynamic> config) {
+  // I have no idea why the static methods needed to be generated.
```

Comment:
Can you elaborate on that? The static mapper methods was a recent feature request (see https://github.com/smotastic/smartstruct/issues/53)
The return type of a method shouldn't be abstract any way. The mapper cannot handle e.g a List as return type anyway. Or am I missing something?

Reply:
You are right, it's my problem. I misunderstand the usage of the static method in the mapper class. I use it for the function mapping just like this:
```dart

@Mapper(static: true)
class ObjectMapper {
    // ...

    static List<Source> mapToList(Source source) {
        return [source];
    }
}
```
Now I realize the function mapping method should write out of the Mapper class, and I also know why the static method should be generated. And I think my code here is not needed.

Static mapping is a good idea. The mapper will be easier to use with the static mapping. But I have another idea to implement it. Why not generate a proxy class just like:
```dart

@Mapper(static: true)
class ObjectMapper {
    Target1 fromSource1(Source1 source);
    Target2 fromSource2(Source2 source);
    Target3 fromSource2(Source3 source);
}


// generate code
class MapperImpl {
    // ...
}

class Mapper$ {
    final static mapper = MapperImpl();

    static Target1 fromSource1(Source1 source) => mapper.fromSource1(Source1 source);
    static Target2 fromSource1(Source2 source) => mapper.fromSource1(Source2 source);
    static Target3 fromSource1(Source3 source) => mapper.fromSource1(Source3 source);
}
// generate code end

```
In this mode, user don't need to write more code for the static method and also can enjoy the static mapping. The mapper code of user will be more clean. 
It just a new feature and has no conflict with the current static mapping feature. I think I will implement it later.

Reply:
I used the idea from the json_serializable library.
The idea is to just generate the implementations, and let the user handle accessing the implementations.
For example by providing your own class, and just calling the generated implementations.

But I can also see advantages in this approach, so as you said it might be implemented later